python/idsse_common/idsse/common/config.py

```diff
@@ -12,17 +12,18 @@
 import glob
 import json
 import logging
+from collections.abc import Iterable
 from inspect import signature
-from typing import Self, Union, List, Optional
+from typing import Self
 
 logger = logging.getLogger(__name__)
 
 
 class Config:
     """Configuration data class"""
     def __init__(self,
-                 config: Union[dict, List[dict], str],
-                 keys: Optional[Union[list, str]] = None,
+                 config: dict | Iterable[dict] | str,
```

Comment:
I thought I read somewhere that Iterable was preferred for arguments because it gives the most flexibility to the caller, whereas Sequence was preferred for return values because it's more specific, and you (hopefully) know what exactly you're returning. 

In general I too prefer Sequence.